src/toil/wdl/wdl_functions.py

```diff
@@ -937,3 +937,15 @@ def wdl_zip(left: List[Any], right: List[Any]) -> List[WDLPair]:
 
     return list(WDLPair(left=left_val, right=right_val) for left_val, right_val in zip(left, right))
 
+
+def cross(left: List[Any], right: List[Any]) -> List[WDLPair]:
+    """
+    Return the cross product of the two arrays. Array[Y][1] appears before
+    Array[X][1] in the output.
+
+    WDL syntax: Array[Pair[X,Y]] cross(Array[X], Array[Y])
+    """
+    if not isinstance(left, list) or not isinstance(right, list):
```

Comment:
Not `WDLArrayType`?

Reply:
No, because `WDLType` isn't a subclass of `type` and `Array[]` is still represented as a `list` in the Python script.

Reply:
You're right, it should be left for a separate PR.